miniapp/miniapp_cholesky.cpp

```diff
@@ -99,8 +99,8 @@ int hpx_main(hpx::program_options::variables_map& vm) {
 
     double gigaflops;
     {
-      double n = matrix.size().rows();
-      double add_mul = n * n * n / 6;
+      auto n = static_cast<std::size_t>(matrix.size().rows());
+      auto add_mul = static_cast<std::size_t>(n * n * n / 6);
       gigaflops = total_ops<T>(add_mul, add_mul) / elapsed_time / 1e9;
```

Comment:
Why `int -> size_t -> double` instead of `int -> double -> double`?

Reply:
`total_ops` accepts `std::size_t` as parameters (being number of additions and multiplications).

So, `n^3 / 6` is cast to an integer, while the output of the `total_ops` is implicitly cast to double when divided by `elapsed_time`.

Reply:
That was part of the question; consider changing total_ops to double to avoid multiple conversions...
moreover max of size_t (~(2M)^3) may be a little bit tight. Currently we plan to run the eigensolver on size 100k matrices, which is just an order of magnitude away from the limit of the number whose cube can be fit in size_t.

Reply:
I changed the code, the function `total_ops` now accepts two `double`. 😉